src/vanilla/Templates/Rest/Common/EnumTemplate.cshtml

```diff
@@ -9,7 +9,50 @@ namespace @(Settings.Namespace).@(Settings.ModelsName)
     /// <summary>
     @WrapComment("/// ", "Defines values for " + Model.Name + ".")
     /// </summary>
-    @if (!Model.ModelAsString)
+    @if((bool)Settings.CustomSettings["ExtensibleEnums"])
+    {
+    @:[Newtonsoft.Json.JsonConverter(typeof(Microsoft.Rest.ClientRuntime.Serialization.ExtensibleEnumConverter<@Model.ClassName>))]
+    @:public sealed class @Model.ClassName : Microsoft.Rest.ExtensibleEnum<@Model.ClassName>
+    @:{
+        @foreach (var t in Model.Values)
+        {
+            if (t.Description != null)
+            {<text>
+        /// <summary>
+        @WrapComment("/// ", t.Description)
+        /// </summary>
+        </text>
+        }
+        @:public static readonly @Model.ClassName @t.Name = new @(Model.ClassName)("@(t.Name)");
+    }
+    @EmptyLine
+        @:private @(Model.ClassName)(string typeName) : base(typeName)
```

Comment:
while we're at it, and since this is actually a highly potential data plane candidate and on our TODO list: we should also have support for non-string enums